packages/assets-controllers/src/TokensController.ts

```diff
@@ -693,11 +696,36 @@ export class TokensController extends BaseControllerV1<
   }
 
   _getProvider(networkClientId?: NetworkClientId): Web3Provider {
-    return new Web3Provider(
-      networkClientId
-        ? this.getNetworkClientById(networkClientId).provider
-        : this.config?.provider,
-    );
+    let provider = this.config?.provider;
+    if (networkClientId) {
+      const networkClientProvider =
+        this.getNetworkClientById(networkClientId).provider;
+      provider = {
+        ...networkClientProvider,
+        send: (
+          _req: { method: string; params?: Json[] },
+          _callback: (error: unknown, response: unknown) => void,
+        ) =>
+          networkClientProvider.send.bind(provider)(
+            { ..._req, id: '', jsonrpc: '2.0' },
+            _callback,
+          ),
+        sendAsync: (
+          _req: { method: string; params?: Json[] },
+          _callback: (error: unknown, response: unknown) => void,
+        ) =>
+          networkClientProvider.sendAsync.bind(provider)(
+            { ..._req, id: '', jsonrpc: '2.0' },
+            _callback,
+          ),
+      };
+    }
+
+    if (provider === undefined) {
+      throw new Error(MISSING_PROVIDER_ERROR);
+    }
+
+    return new Web3Provider(provider);
```

Comment:
Similar point as for AssetsContractController — maybe we don't worry about the wrapper here and we use a `@ts-expect-error`?

``` suggestion
    const provider = networkClientId
      ? this.getNetworkClientById(networkClientId).provider
      : this.config.provider;

    if (!provider) {
      throw new Error(MISSING_PROVIDER_ERROR);
    }

    // @ts-expect-error TODO: remove this annotation once the `Eip1193Provider` class is released
    return new Web3Provider(provider);
```